zulipterminal/model.py

```diff
@@ -1517,6 +1517,13 @@ def notify_user(self, message: Message) -> str:
             ) or self.is_visual_notifications_enabled(stream_id):
                 recipient = "{display_recipient} -> {subject}".format(**message)
 
+            if "has_alert_word" in message["flags"]:
+                # Check if stream or topic is muted
+                topic = message.get("subject", "")
+                if not self.is_muted_stream(stream_id) and not self.is_muted_topic(
+                    stream_id, topic
+                ):
+                    recipient = "{display_recipient} -> {subject}".format(**message)
```

Comment:
See my response to a comment on an earlier iteration of the PR regarding simplifying this.

Reply:
How about creating a new function for providing value of `recipient`  in the notification?
We have 2 conditionals: private and stream message. Under stream message, the code for assigning value to `recipient` requires us to check across multiple conditions. One way to simplify this entire block can be to create a separate function which returns the value of `recipient`. This can also be used for assigning value to recipient under the private message conditional as well.

We can have the `message` as a parameter passed to the new function. Also, can we have this as an nested function?

Reply:
The original simplification I saw possible here was the equivalent of changing
```python
if big_long_complicated_expression_for_one_case:
  x = value
if another_big_long_complicated_expression_for_second_case:
  x = value
```
into
```python
is_one_case = big_long_complicated_expression_for_one_case
is_second_case = big_long_complicated_expression_for_second_case
if is_one_case or is_second_case:
  x = value
```
This has the benefit of easily naming the two conditions one is testing for, via the intermediate variables, and deduplicating the assignment.

Reply:
Oh alright. This is a simpler approach than what I had suggested.